changelog.d/3065.change.rst

```diff
@@ -0,0 +1,4 @@
+Refactored ``setuptools.config`` by separating configuration parsing (specific
```

Comment:
Is this change user-facing? Maybe change the fragment type to `misc` if it's internal?